src/devices/Media/SoundDevice/Devices/UnixSoundDevice.cs

```diff
@@ -177,114 +207,156 @@ public override void Record(uint recordTimeSeconds, Stream outputStream)
                 SubChunk2 = subChunk2
             };
 
-            WriteWavHeader(outputStream, header);
+            return header;
+        }
 
-            try
+        public override void StartRecording(string outputFilePath)
+        {
+            Record(outputFilePath);
+        }
+
+        public override void StopRecording()
+        {
+            _record = false;
+            while (_recordingThread.IsAlive)
```

Comment:
should you just `_recordingThread.Join()`

Reply:
yes, that should work as well.